### PR TITLE
Ticket #48: Recette version 4.2 de Shopping Flux

### PR DESCRIPTION
**Pull request by Vincent-Profileo**
Fixes # 47

Changes made
Ticket #41: Clean of warnings and notices
Ticket #31: Stability of breadcrumb in case of incoherence in categories
Ticket #44: Cdiscount fees
Ticket #37: Active boolean on SQL queries
Ticket #38: Log system

### CODE REVIEW

File: shoppingfluxexport.php [hunk 1]
```diff
@@ -788,24 +793,74 @@ public function writeFeed($total, $current = 0)
...
+            
+            // Remove previous feed an place the newly generated one
+            $shop_id = $this->context->shop->id;
+            unlink(dirname(__FILE__).'/feed_'.$shop_id.'.xml');
+            rename(dirname(__FILE__).'/feed_'.$shop_id.'_tmp.xml', dirname(__FILE__).'/feed_'.$shop_id.'.xml');
+            
+            // Notify end of cron execution
+            $this->logDebug('EXPORT SUCCESSFULL');
+            $curl_post_data = array();
+            $uri = 'http://www.shopping-flux.com/';
```
Comment (ronan-gloo) [inline shoppingfluxexport.php]: Uses https instead

File: shoppingfluxexport.php [hunk 2]
```diff
@@ -788,24 +793,74 @@ public function writeFeed($total, $current = 0)
...
-            fclose($log);
-        }
-
+       
         if ($current + $configuration['PASSES'] >= $total) {
             $this->closeFeed();
+            
+            // Remove previous feed an place the newly generated one
+            $shop_id = $this->context->shop->id;
+            unlink(dirname(__FILE__).'/feed_'.$shop_id.'.xml');
```
Comment (ronan-gloo) [inline shoppingfluxexport.php]: This will trigger a warning if file does not exists

File: shoppingfluxexport.php [hunk 3]
```diff
@@ -559,128 +611,65 @@ public function generateFeed()
...
-                    echo '</discount>';
-                }
-                echo '</discounts>';
+        // Filename as the shop id so we can generate a feed by shop
+        $shop_id = $this->context->shop->id;
+        $filename = dirname(__FILE__).'/feed_'.$shop_id.'.xml';
+        
+        // Check if file exists
+        if (!file_exists($filename)) {
+            p('XML file does not exists');
```
Comment (ronan-gloo) [inline shoppingfluxexport.php]: Shopping-Flux is expecting a XML response, `p()` function provides legacy HTML.

**Comment by flebap** [comment]
We have made changes to this PR and have it re-published : https://github.com/shoppingflux/shoppingfluxexport/pull/52

We remove the cURL and the 6 hours restrictions on the Cron.

**Pull request closed without merging**